File: svws-webclient/core/src/core/data/uv/UvLerngruppenLehrer.ts
import { JavaLong } from '../../../java/lang/JavaLong';
import { JavaObject } from '../../../java/lang/JavaObject';
import { Class } from '../../../java/lang/Class';

export class UvLerngruppenLehrer extends JavaObject {

	/**
	 * Die ID der Zuordnung (planungsspezifisch).
	 */
	public id: number = -1;

	/**
	 * Die ID des Planungsabschnitts, in dem die Zuordnung gilt.
	 */
	public idPlanungsabschnitt: number = -1;

	/**
	 * Die ID der Lerngruppe, der der Lehrer zugeordnet ist.
	 */
	public idLerngruppe: number = -1;

	/**
	 * Die ID des Lehrers, der der Lerngruppe zugeordnet ist.
	 */
	public idLehrer: number = -1;

	/**
	 * Die Reihenfolge der Zuordnung (z. B. 1 = Hauptlehrkraft, 2 = Zweitkraft).
	 */
	public reihenfolge: number = 1;

	/**
	 * Die Anzahl der Wochenstunden, in denen der Lehrer in dieser Lerngruppe eingesetzt ist.
	 */
	public wochenstunden: number = 0.0;

	/**
	 * Die Anzahl der Wochenstunden, die auf das Deputat angerechnet werden.
	 */
	public wochenstundenAngerechnet: number = 0.0;


	/**
	 * Leerer Standardkonstruktor.
	 */
	public constructor() {
		super();
	}

	/**
	 * Vergleicht, ob das aktuelle Objekt dasselbe ist wie ein anderes übergebenes Objekt.
	 *
	 * @param another das zu vergleichende Objekt
	 * @return true, falls die Objekte identisch sind, sonst false
	 */
	public equals(another: unknown | null): boolean {
		return (((another instanceof JavaObject) && (another.isTranspiledInstanceOf('de.svws_nrw.core.data.uv.UvLerngruppenLehrer')))) && (this.id === (another as unknown as UvLerngruppenLehrer).id);
	}

	/**
	 * Erzeugt den Hashcode zu Objekt auf Basis der ID.
	 *
	 * @return den HashCode
	 */
	public hashCode(): number {
		return JavaLong.hashCode((this.id));
	}

	/**
	 * Gibt eine String-Repräsentation des Objekts zurück.
	 *
	 * @return die String-Darstellung der Lehrer-Lerngruppen-Zuordnung
	 */
	public toString(): string | null {
		return this.idLerngruppe + "-" + this.idLehrer;
	}

	transpilerCanonicalName(): string {
		return 'de.svws_nrw.core.data.uv.UvLerngruppenLehrer';
	}

	isTranspiledInstanceOf(name: string): boolean {
		return ['de.svws_nrw.core.data.uv.UvLerngruppenLehrer'].includes(name);
	}

	public static readonly class = new Class<UvLerngruppenLehrer>('de.svws_nrw.core.data.uv.UvLerngruppenLehrer');

	public static transpilerFromJSON(json: string): UvLerngruppenLehrer {
		const obj = JSON.parse(json) as Partial<UvLerngruppenLehrer>;
		const result = new UvLerngruppenLehrer();
		if (obj.id === undefined)
			throw new Error('invalid json format, missing attribute id');
		result.id = obj.id;
		if (obj.idPlanungsabschnitt === undefined)
			throw new Error('invalid json format, missing attribute idPlanungsabschnitt');
		result.idPlanungsabschnitt = obj.idPlanungsabschnitt;
		if (obj.idLerngruppe === undefined)
			throw new Error('invalid json format, missing attribute idLerngruppe');
		result.idLerngruppe = obj.idLerngruppe;
		if (obj.idLehrer === undefined)
			throw new Error('invalid json format, missing attribute idLehrer');
		result.idLehrer = obj.idLehrer;
		if (obj.reihenfolge === undefined)
			throw new Error('invalid json format, missing attribute reihenfolge');
		result.reihenfolge = obj.reihenfolge;
		if (obj.wochenstunden === undefined)
			throw new Error('invalid json format, missing attribute wochenstunden');
		result.wochenstunden = obj.wochenstunden;
		if (obj.wochenstundenAngerechnet === undefined)
			throw new Error('invalid json format, missing attribute wochenstundenAngerechnet');
		result.wochenstundenAngerechnet = obj.wochenstundenAngerechnet;
		return result;
	}

	public static transpilerToJSON(obj: UvLerngruppenLehrer): string {
		let result = '{';
		result += '"id" : ' + obj.id.toString() + ',';
		result += '"idPlanungsabschnitt" : ' + obj.idPlanungsabschnitt.toString() + ',';
		result += '"idLerngruppe" : ' + obj.idLerngruppe.toString() + ',';
		result += '"idLehrer" : ' + obj.idLehrer.toString() + ',';
		result += '"reihenfolge" : ' + obj.reihenfolge.toString() + ',';
		result += '"wochenstunden" : ' + obj.wochenstunden.toString() + ',';
		result += '"wochenstundenAngerechnet" : ' + obj.wochenstundenAngerechnet.toString() + ',';
		result = result.slice(0, -1);
		result += '}';
		return result;
	}

	public static transpilerToJSONPatch(obj: Partial<UvLerngruppenLehrer>): string {
		let result = '{';
		if (obj.id !== undefined) {
			result += '"id" : ' + obj.id.toString() + ',';
		}
		if (obj.idPlanungsabschnitt !== undefined) {
			result += '"idPlanungsabschnitt" : ' + obj.idPlanungsabschnitt.toString() + ',';
		}
		if (obj.idLerngruppe !== undefined) {
			result += '"idLerngruppe" : ' + obj.idLerngruppe.toString() + ',';
		}
		if (obj.idLehrer !== undefined) {
			result += '"idLehrer" : ' + obj.idLehrer.toString() + ',';
		}
		if (obj.reihenfolge !== undefined) {
			result += '"reihenfolge" : ' + obj.reihenfolge.toString() + ',';
		}
		if (obj.wochenstunden !== undefined) {
			result += '"wochenstunden" : ' + obj.wochenstunden.toString() + ',';
		}
		if (obj.wochenstundenAngerechnet !== undefined) {
			result += '"wochenstundenAngerechnet" : ' + obj.wochenstundenAngerechnet.toString() + ',';
		}
		result = result.slice(0, -1);
		result += '}';
		return result;
	}

}

export function cast_de_svws_nrw_core_data_uv_UvLerngruppenLehrer(obj: unknown): UvLerngruppenLehrer {
	return obj as UvLerngruppenLehrer;
}

File: svws-core/src/main/java/de/svws_nrw/core/data/uv/UvLerngruppenLehrer.java
package de.svws_nrw.core.data.uv;

import de.svws_nrw.transpiler.TranspilerDTO;
import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.xml.bind.annotation.XmlRootElement;

/**
 * Diese Klasse wird bei der Kommunikation über die Open-API-Schnittstelle verwendet.
 * Sie liefert die Informationen zur Zuordnung eines Lehrers zu einer Lerngruppe
 * innerhalb eines Planungsabschnitts der Unterrichtsverteilung.
 */
@XmlRootElement
@Schema(description = "die Informationen zur Zuordnung eines Lehrers zu einer Lerngruppe innerhalb eines Planungsabschnitts.")
@TranspilerDTO
public class UvLerngruppenLehrer {

	/** Die ID der Zuordnung (planungsspezifisch). */
	@Schema(description = "die ID der Zuordnung (planungsspezifisch)", example = "4711")
	public long id = -1;

	/** Die ID des Planungsabschnitts, in dem die Zuordnung gilt. */
	@Schema(description = "die ID des Planungsabschnitts, in dem die Zuordnung gilt", example = "101")
	public long idPlanungsabschnitt = -1;

	/** Die ID der Lerngruppe, der der Lehrer zugeordnet ist. */
	@Schema(description = "die ID der Lerngruppe, der der Lehrer zugeordnet ist", example = "2001")
	public long idLerngruppe = -1;

	/** Die ID des Lehrers, der der Lerngruppe zugeordnet ist. */
	@Schema(description = "die ID des Lehrers, der der Lerngruppe zugeordnet ist", example = "3001")
	public long idLehrer = -1;

	/** Die Reihenfolge der Zuordnung (z. B. 1 = Hauptlehrkraft, 2 = Zweitkraft). */
	@Schema(description = "die Reihenfolge der Zuordnung (z. B. 1 = Hauptlehrkraft, 2 = Zweitkraft)", example = "1")
	public int reihenfolge = 1;

	/** Die Anzahl der Wochenstunden, in denen der Lehrer in dieser Lerngruppe eingesetzt ist. */
	@Schema(description = "die Anzahl der Wochenstunden, in denen der Lehrer in dieser Lerngruppe eingesetzt ist", example = "4.0")
	public double wochenstunden = 0.0;

	/** Die Anzahl der Wochenstunden, die auf das Deputat angerechnet werden. */
	@Schema(description = "die Anzahl der Wochenstunden, die auf das Deputat angerechnet werden", example = "2.0")
	public double wochenstundenAngerechnet = 0.0;

	/**
	 * Leerer Standardkonstruktor.
	 */
	public UvLerngruppenLehrer() {
		// leer
	}

	/**
	 * Gibt eine String-Repräsentation des UvLerngruppenLehrer-Objekts zurück.
	 *
	 * @return die String-Darstellung der Lehrer-Lerngruppen-Zuordnung
	 */
	@Override
	public String toString() {
		return "UvLerngruppenLehrer{id=" + id
				+ ", idPlanungsabschnitt=" + idPlanungsabschnitt
				+ ", idLerngruppe=" + idLerngruppe
				+ ", idLehrer=" + idLehrer
				+ ", reihenfolge=" + reihenfolge
				+ ", wochenstunden=" + wochenstunden
				+ ", wochenstundenAngerechnet=" + wochenstundenAngerechnet
				+ "}";
	}

	/**
	 * Vergleicht, ob das aktuelle Objekt dasselbe ist wie ein anderes übergebenes Objekt.
	 *
	 * @param another das zu vergleichende Objekt
	 * @return true, falls die Objekte identisch sind, sonst false
	 */
	@Override
	public boolean equals(final Object another) {
		return (another instanceof final UvLerngruppenLehrer a)
				&& (this.id == a.id);
	}

	/**
	 * Erzeugt den Hashcode zu Objekt auf Basis der ID.
	 *
	 * @return den HashCode
	 */
	@Override
	public int hashCode() {
		return Long.hashCode(id);
	}

}
